lms/static/scripts/frontend_apps/components/Dialog.js

```diff
@@ -42,48 +43,37 @@ export default function Dialog({
   title,
   buttons,
 }) {
-  const handleKey = event => {
-    event.stopPropagation();
-
-    if (event.key === 'Escape' && typeof onCancel === 'function') {
+  const contentEl = useRef(null);
+  useElementShouldClose(contentEl, true, () => {
+    // onCancel callback prop is optional
+    if (typeof onCancel === 'function') {
       onCancel();
     }
-  };
+  });
 
-  const rootEl = useRef(null);
   useEffect(() => {
     if (initialFocus) {
       initialFocus.current.focus();
-    } else {
-      // Modern accessibility guidance is to focus the dialog itself rather than
-      // trying to be smart about focusing a particular control within the
-      // dialog. See resources above.
-      rootEl.current.focus();
     }
-
     // We only want to run this effect once when the dialog is mounted.
     //
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
   return (
-    // FIXME-A11Y
-    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
     <div
-      role="dialog"
+      role="alertdialog"
```

Comment:
The `Dialog` control is used in multiple contexts, some of which `alertdialog` is appropriate for (eg. `ErrorDialog`) but many where I think `dialog` would make more sense (eg. the dialogs that appear in the file picker app when selecting a source to pick a file from). I'd suggest that we make this a configurable prop.

What are the effects of using `alertdialog` vs `dialog`?